lib/tools/settings-client-commands.js

```diff
@@ -39,8 +39,34 @@ const GPS_CACHE_REFRESHED_LOGS = [
 
 const GPS_COORDINATES_PATTERN = /data="(-?[\d.]+)\s+(-?[\d.]+)\s+(-?[\d.]+)"/;
 
+const FORE_GROUND_APP_KEYWORD = 'isForeground=true';
+
 const commands = {};
 
+/**
+ * If the io.appium.settings package has running foreground service.
+ * It returns the io.appium.settings's process existence for api level 25 and lower
+ * since it does not have the foreground service.
+ *
+ * @this {import('../adb.js').ADB}
+ * @returns {Promise<boolean>} Return true if the device has running settings app foreground service.
+ */
+commands.hasRunningSettingsAppForegroundService = async function hasRunningSettingsAppForegroundService () {
+  if (await this.getApiLevel() < 26) {
+    return await this.processExists(SETTINGS_HELPER_ID);
+  }
+
+  // The foreground service is available since api level 26,
+  // thus this method works only for api level 26+.
+  try {
+    const output = await this.getActivityService(SETTINGS_HELPER_ID);
+    return (output.includes(FORE_GROUND_APP_KEYWORD));
+  } catch (e) {
+    log.warn(`Got an error in getting the foreground service state: ${e.message}`);
```

Comment:
let it throw an error - we should know when it does not work

Reply:
https://github.com/appium/appium-adb/pull/715/commits/ddf5635a247c4896dd19c703a13c11f567165b84